From 5e468ee64655b1ff446d88667bfac2f6469e7b12 Mon Sep 17 00:00:00 2001
From: "evelyn.tsai" <evelyn.tsai@mediatek.com>
Date: Tue, 4 Mar 2025 16:50:29 +0800
Subject: [PATCH 183/185] Revert "mtk: hostapd: make sure all BSSes are MLD
 enabled when phy mode is BE"

This reverts commit 3dcee3516e8ddf327377bcd5585be912498e340d.

Reason for revert: unstable single link mld

---
 src/ap/ap_config.c | 10 ----------
 src/ap/hostapd.h   |  2 +-
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index a9344899b..f636b43d3 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1661,16 +1661,6 @@ int hostapd_config_check(struct hostapd_config *conf, int full_config)
 	if (full_config)
 		hostapd_set_and_check_bw320_offset(conf,
 						   conf->eht_bw320_offset);
-
-	/* Make sure that all BSSes are MLD enabled when ieee80211be is set */
-	if (full_config && conf->ieee80211be) {
-		for (i = 0; i < conf->num_bss; i++) {
-			if (!conf->bss[i]->mld_ap) {
-				conf->bss[i]->mld_ap = 1;
-				conf->bss[i]->mld_allowed_links = 1;
-			}
-		}
-	}
 #endif /* CONFIG_IEEE80211BE */
 
 	if (full_config && conf->mbssid && !conf->ieee80211ax) {
diff --git a/src/ap/hostapd.h b/src/ap/hostapd.h
index 56f703d69..e31f6c7e3 100644
--- a/src/ap/hostapd.h
+++ b/src/ap/hostapd.h
@@ -1100,7 +1100,7 @@ static inline u32 hostapd_get_radio_mask(struct hostapd_data *hapd)
 {
 #ifdef CONFIG_IEEE80211BE
 	if (hapd->iface->current_hw_info) {
-		if (hapd->conf->mld_ap && hapd->conf->mld_radio_mask)
+		if (hapd->conf->mld_ap)
 			return hapd->conf->mld_radio_mask;
 		else
 			return 1 << hapd->iface->current_hw_info->hw_idx;
-- 
2.45.2

